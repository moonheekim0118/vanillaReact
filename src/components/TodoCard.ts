import { todoObject } from "../util/types";

const TodoCard = () => {
    function render({ id, description, done }: todoObject): Element {
        const dataId = id.toString();
        const TodoCard = document.createElement("div");
        TodoCard.className = "Todo-Card";
        TodoCard.dataset.id = dataId;

        const TodoDesc = document.createElement("span");
        TodoDesc.className = "Todo-Descr";
        TodoDesc.innerText = description;
        TodoDesc.dataset.id = dataId;

        const TodoDone = document.createElement("i");
        TodoDone.className = done ? "fas fa-check" : "";
        TodoDone.dataset.id = dataId;

        const RemoveIcon = document.createElement("i");
        RemoveIcon.className = "fas fa-trash-alt";
        RemoveIcon.id = "Remove-Btn";
        RemoveIcon.dataset.id = dataId;

        TodoCard.appendChild(TodoDesc);
        TodoCard.appendChild(TodoDone);
        TodoCard.appendChild(RemoveIcon);

        return TodoCard;
    }

    return render;
};

export default TodoCard();
