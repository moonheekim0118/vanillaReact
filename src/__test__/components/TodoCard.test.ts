import "jsdom-global/register";
import TodoCard from "../../components/TodoCard";

describe("<TodoCard/>", () => {
    let container;
    const MockData = {
        id: 1,
        description: "test",
        done: true
    };

    it("renders correctly", () => {
        container = TodoCard(MockData);
    });

    it("matches snapshot", () => {
        expect(container).toMatchSnapshot();
    });
});
