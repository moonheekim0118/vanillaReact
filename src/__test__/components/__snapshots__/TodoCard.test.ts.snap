// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoCard/> matches snapshot 1`] = `
<div
  class="Todo-Card done"
  data-id="1"
  draggable="true"
>
  <span
    class="Todo-Descr"
    data-id="1"
  />
  <i
    class="fas fa-check"
    data-id="1"
  />
  <i
    class="fas fa-trash-alt"
    data-id="1"
    id="Remove-Btn"
  />
</div>
`;
